community/cypher/cypher/src/main/java/org/neo4j/cypher/javacompat/internal/GraphDatabaseCypherService.java

```diff
@@ -19,28 +19,31 @@
  */
 package org.neo4j.cypher.javacompat.internal;
 
-import java.net.URL;
-import java.util.concurrent.TimeUnit;
-
-import org.neo4j.graphdb.DependencyResolver;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Label;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.*;
 import org.neo4j.graphdb.security.URLAccessValidationError;
 import org.neo4j.kernel.GraphDatabaseQueryService;
 import org.neo4j.kernel.api.KernelTransaction;
+import org.neo4j.kernel.api.dbms.DbmsOperations;
 import org.neo4j.kernel.api.security.SecurityContext;
+import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge;
 import org.neo4j.kernel.impl.coreapi.InternalTransaction;
 import org.neo4j.kernel.impl.factory.GraphDatabaseFacade;
 
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
 public class GraphDatabaseCypherService implements GraphDatabaseQueryService
 {
-    private GraphDatabaseFacade graph;
+    private final GraphDatabaseFacade graph;
+    private final DbmsOperations dbmsOperations;
+    private final ThreadToStatementContextBridge txBridge;
 
     public GraphDatabaseCypherService( GraphDatabaseService graph )
     {
         this.graph = (GraphDatabaseFacade) graph;
+        DependencyResolver resolver = getDependencyResolver();
+        this.dbmsOperations = resolver.resolveDependency( DbmsOperations.class );
```

Comment:
Why do we need to use `GraphDatabaseCypherService` as a service provider instead of dependencyResolver?
I have tried to find a place where we call `getDbmsOperations` or `getTxBridge` to get stored services and it looks like we always have dependencyResolver available there.